types/validator_set_test.go

```diff
@@ -658,6 +651,60 @@ type testVal struct {
 	power int64
 }
 
+func permutation(valList []testVal) []testVal {
+	if len(valList) == 0 {
+		return nil
+	}
+	permList := make([]testVal, len(valList))
+	//perm := rand.Perm(len(valList))
+	perm := cmn.RandPerm(len(valList))
+	for i, v := range perm {
+		permList[v] = valList[i]
+	}
+	return permList
+}
+
+func createNewValidatorList(testValList []testVal) []*Validator {
```

Comment:
why do we use testVal? can't we use `Validator` struct?

Reply:
I find it easier to write the test values this way and avoid the casting.